src/libraries/System.Net.Sockets/src/System/Net/Sockets/UDPClient.cs

```diff
@@ -618,6 +656,39 @@ public Task<int> SendAsync(byte[] datagram, int bytes, IPEndPoint? endPoint)
             }
         }
 
+        /// <summary>
+        /// Sends a UDP datagram asynchronously to a remote host.
+        /// </summary>
+        /// <param name="datagram">
+        /// An <see cref="ReadOnlyMemory{T}"/> of Type <see cref="byte"/> that specifies the UDP datagram that you intend to send.
+        /// </param>
+        /// <param name="endPoint">
+        /// An <see cref="IPEndPoint"/> that represents the host and port to which to send the datagram.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// The token to monitor for cancellation requests. The default value is None.
+        /// </param>
+        /// <returns>A <see cref="ValueTask{T}"/> that represents the asynchronous send operation. The value of its Result property contains the number of bytes sent.</returns>
+        /// <exception cref="InvalidOperationException"><see cref="UdpClient"/> has already established a default remote host and <paramref name="endPoint"/> is not <see langword="null"/>.</exception>
+        /// <exception cref="ObjectDisposedException">The <see cref="UdpClient"/> is closed.</exception>
+        /// <exception cref="SocketException">An error occurred when accessing the socket.</exception>
+        public ValueTask<int> SendAsync(ReadOnlyMemory<byte> datagram, IPEndPoint? endPoint, CancellationToken cancellationToken = default)
+        {
+            ThrowIfDisposed();
+
+            if (endPoint is null)
+            {
+                return _clientSocket.SendAsync(datagram, SocketFlags.None, cancellationToken);
+            }
+            if (_active)
+            {
+                // Do not allow sending packets to arbitrary host when connected.
+                throw new InvalidOperationException(SR.net_udpconnected);
```

Comment:
> the existing method public Task SendAsync(byte[] datagram, int bytes) did not check if current udpClient is not connected but existing sync version checked.

We should open an issue about this, but we should not change behavior in this PR.

> The new async version keep same as public Task<int> SendAsync(byte[] datagram, int bytes).

It looks like you throw an exception at the lines marked, I assume it's a leftover by mistake.

Reply:
This method is paired for existing `public Task<int> SendAsync(byte[] datagram, int bytes, IPEndPoint? endPoint);`, not mentioned existing `public Task SendAsync(byte[] datagram, int bytes)`. Here the method code is refactored to inline some verification code as previous comment, so it is a little different than existing one (but behavior should be same). 
This exception marked is for when user provides valid endPoint but udpClient is already connected. (existing pair method also did that check)